src/http/common.cc

```diff
@@ -134,6 +134,48 @@ output_stream<char> make_http_chunked_output_stream(output_stream<char>& out) {
     return output_stream<char>(http_chunked_data_sink(out), default_body_sink_buffer_size, opts);
 }
 
+class http_content_length_data_sink_impl : public data_sink_impl {
+    output_stream<char>& _out;
+    const size_t _limit;
+    size_t& _bytes_written;
+
+public:
+    http_content_length_data_sink_impl(output_stream<char>& out, size_t& len)
+            : _out(out)
+            , _limit(std::exchange(len, 0))
+            , _bytes_written(len)
+    {
+    }
+    virtual future<> put(net::packet data)  override { abort(); }
```

Comment:
What's this? :-(

Reply:
```
class data_sink_impl {
public:
    virtual future<> put(net::packet data) = 0;
```

inheriting classes must implement this method, but since http client doesn't put net::packet-s into it it's empty this way. Do you think it should look some other way instead? The http_chunked_data_sink_impl is the same in this sense

Reply:
I guess that if this sink class is not public, and its user never uses the put(net::packet) method, it's ok that we do there something silly like abort. So, fine.

We could have had in data_sink_impl a default put(net::packet) implementation which calls the other put() functions, to avoid this ugliness, but of course this is not in scope for this patch.